CCDM: commons-lang3 conflict when building project with 15.0.0.alpha4

Building using `mvn install -Pproduction` with 15.0.0.alpha4 results in
```
[ERROR] Failed to execute goal com.vaadin:vaadin-maven-plugin:15.0.0.alpha4:build-frontend (default) on project studio: Execution default of goal com.vaadin:vaadin-maven-plugin:15.0.0.alpha4:build-frontend failed: An API incompatibility was encountered while executing com.vaadin:vaadin-maven-plugin:15.0.0.alpha4:build-frontend: java.lang.NoSuchMethodError: org.apache.commons.lang3.StringUtils.firstNonBlank([Ljava/lang/CharSequence;)Ljava/lang/CharSequence;
```

Using the same versions but `flow-maven-plugin` `3.0.0.alpha7` instead of `vaadin-maven-plugin 15.0.0.alpha4` works fine

--- a/flow-maven-plugin/pom.xml
+++ b/flow-maven-plugin/pom.xml
@@ -77,10 +77,6 @@
             <scope>test</scope>
         </dependency>
         <dependency>
-            <groupId>org.apache.commons</groupId>
-            <artifactId>commons-lang3</artifactId>
-        </dependency>
-        <dependency>
             <groupId>org.sonatype.plexus</groupId>
             <artifactId>plexus-build-api</artifactId>
             <version>0.0.7</version>

--- a/flow-maven-plugin/pom.xml
+++ b/flow-maven-plugin/pom.xml
@@ -76,7 +76,10 @@
             <artifactId>slf4j-simple</artifactId>
             <scope>test</scope>
         </dependency>
-
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-lang3</artifactId>
+        </dependency>
         <dependency>
             <groupId>org.sonatype.plexus</groupId>
             <artifactId>plexus-build-api</artifactId>

--- a/flow-server/pom.xml
+++ b/flow-server/pom.xml
@@ -249,11 +249,7 @@
                 </exclusion>
             </exclusions>
         </dependency>
-        <!-- Need to be revised and removed to avoid conflicts-->
-        <dependency>
-            <groupId>org.apache.commons</groupId>
-            <artifactId>commons-lang3</artifactId>
-        </dependency>
+
         <!-- Connect Testing dependendies -->
 
         <!-- TODO: re-enable tests mocking final classes in the connect

--- a/flow-server/src/main/java/com/vaadin/flow/server/connect/generator/GeneratorUtils.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/connect/generator/GeneratorUtils.java
@@ -1,0 +1,101 @@
+/*
+ * Copyright 2000-2019 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.server.connect.generator;
+
+import java.util.Arrays;
+
+/**
+ * A set of static methods used in CCDM generators, so as flow do not depend on
+ * external libraries for these operations.
+ */
+final class GeneratorUtils {
+    
+    private GeneratorUtils() {
+    }
+
+    static boolean equals(String s1, String s2) {
+        return s1 == s2 || s1 != null && s1.equals(s2);
+    }
+
+    static int compare(String s1, String s2) {
+        return equals(s1, s2) ? 0
+                : s1 == null ? -1 : s2 == null ? 1 : s1.compareTo(s2);
+    }
+
+    static String capitalize(String s) {
+        return s == null ? s : s.substring(0, 1).toUpperCase() + s.substring(1);
+    }
+
+    static boolean isBlank(String s) {
+        return s == null || s.replaceAll("\\s+","").isEmpty();
+    }
+
+    static boolean isNotBlank(String s) {
+        return !isBlank(s);
+    }
+
+    static String firstNonBlank(String... values) {
+        return Arrays.stream(values).filter(GeneratorUtils::isNotBlank)
+                .findFirst().orElse(null);
+    }
+
+    static boolean isTrue(Boolean b) {
+        return Boolean.TRUE.equals(b);
+    }
+
+    static boolean isNotTrue(Boolean b) {
+        return !isTrue(b);
+    }
+
+    static <T> T defaultIfNull(T o, T def) {
+        return o != null ? o : def;
+    }
+    
+    static String replaceChars(String s, char c1, final char c2) {
+        return s == null ? s : s.replace(c1, c2);
+    }
+    
+    @SuppressWarnings("squid:S2259")
+    static boolean contains(String s, String p) {
+        return isNotBlank(s) && isNotBlank(p) && s.contains(p);
+    }
+
+    @SuppressWarnings("squid:S2259")
+    static String substringAfter(String s, String p) {
+        return contains(s, p) ? s.substring(s.indexOf(p) + p.length()) : "";
+    }
+
+    @SuppressWarnings("squid:S2259")
+    static String substringAfterLast(String s, String p) {
+        return contains(s, p) ? s.substring(s.lastIndexOf(p) + p.length()) : "";
+    }
+
+    @SuppressWarnings("squid:S2259")
+    static String substringBeforeLast(String s, String p) {
+        return contains(s, p) ? s.substring(0, s.lastIndexOf(p)) : s;
+    }
+
+    @SuppressWarnings("squid:S2259")
+    static boolean endsWith(String s, String p) {
+        return contains(s, p) && s.length() == p.length() + s.lastIndexOf(p);
+    }
+
+    @SuppressWarnings("squid:S2259")
+    static String removeEnd(String s, String p) {
+        return endsWith(s, p) ? s.substring(0, s.lastIndexOf(p)) : s;
+    }
+
+}

--- a/flow-server/src/main/java/com/vaadin/flow/server/connect/generator/OpenApiObjectGenerator.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/connect/generator/OpenApiObjectGenerator.java
@@ -88,8 +88,6 @@ import io.swagger.v3.oas.models.security.SecurityRequirement;
 import io.swagger.v3.oas.models.security.SecurityScheme;
 import io.swagger.v3.oas.models.servers.Server;
 import io.swagger.v3.oas.models.tags.Tag;
-import org.apache.commons.lang3.BooleanUtils;
-import org.apache.commons.lang3.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
@@ -314,7 +312,7 @@ public class OpenApiObjectGenerator {
                 .map(SingleMemberAnnotationExpr::getMemberValue)
                 .map(Expression::asStringLiteralExpr)
                 .map(LiteralStringValueExpr::getValue)
-                .filter(StringUtils::isNotBlank)
+                .filter(GeneratorUtils::isNotBlank)
                 .orElse(classDeclaration.getNameAsString());
 
         String validationError = serviceNameChecker.check(serviceName);
@@ -375,7 +373,7 @@ public class OpenApiObjectGenerator {
                 typeDeclaration);
         schema.properties(properties);
         List<String> requiredList = properties.entrySet().stream()
-                .filter(stringSchemaEntry -> BooleanUtils
+                .filter(stringSchemaEntry -> GeneratorUtils
                         .isNotTrue(stringSchemaEntry.getValue().getNullable()))
                 .map(Map.Entry::getKey).collect(Collectors.toList());
         // Nullable is represented in requiredList instead.
@@ -410,7 +408,7 @@ public class OpenApiObjectGenerator {
                         variableDeclarator.getType(),
                         fieldDescription.orElse(""));
                 if (field.isAnnotationPresent(Nullable.class)
-                        || BooleanUtils.isTrue(propertySchema.getNullable())) {
+                        || GeneratorUtils.isTrue(propertySchema.getNullable())) {
                     // Temporarily set nullable to indicate this property is
                     // not required
                     propertySchema.setNullable(true);
@@ -425,9 +423,9 @@ public class OpenApiObjectGenerator {
     private Map<String, ResolvedReferenceType> collectUsedTypesFromSchema(
             Schema schema) {
         Map<String, ResolvedReferenceType> map = new HashMap<>();
-        if (StringUtils.isNotBlank(schema.getName())
-                || StringUtils.isNotBlank(schema.get$ref())) {
-            String name = StringUtils.firstNonBlank(schema.getName(),
+        if (GeneratorUtils.isNotBlank(schema.getName())
+                || GeneratorUtils.isNotBlank(schema.get$ref())) {
+            String name = GeneratorUtils.firstNonBlank(schema.getName(),
                     schemaResolver.getSimpleRef(schema.get$ref()));
             ResolvedReferenceType resolvedReferenceType = schemaResolver
                     .getFoundTypeByQualifiedName(name);
@@ -614,12 +612,12 @@ public class OpenApiObjectGenerator {
             usedTypes.putAll(collectUsedTypesFromSchema(paramSchema));
             String name = (isReservedWord(parameter.getNameAsString()) ? "_"
                     : "").concat(parameter.getNameAsString());
-            if (StringUtils.isBlank(paramSchema.get$ref())) {
+            if (GeneratorUtils.isBlank(paramSchema.get$ref())) {
                 paramSchema.description(
                         paramsDescription.remove(parameter.getNameAsString()));
             }
             requestSchema.addProperties(name, paramSchema);
-            if (BooleanUtils.isNotTrue(paramSchema.getNullable())
+            if (GeneratorUtils.isNotTrue(paramSchema.getNullable())
                     && !parameter.isAnnotationPresent(Nullable.class)) {
                 requestSchema.addRequiredItem(name);
             }
@@ -636,7 +634,7 @@ public class OpenApiObjectGenerator {
     private Schema parseTypeToSchema(Type javaType, String description) {
         try {
             Schema schema = parseResolvedTypeToSchema(javaType.resolve());
-            if (StringUtils.isNotBlank(description)) {
+            if (GeneratorUtils.isNotBlank(description)) {
                 schema.setDescription(description);
             }
             return schema;
@@ -801,13 +799,13 @@ public class OpenApiObjectGenerator {
                 .getTypeDeclaration();
         String packageName = typeDeclaration.getPackageName();
         String canonicalName = typeDeclaration.getQualifiedName();
-        if (StringUtils.isBlank(packageName)) {
-            return StringUtils.replaceChars(canonicalName, '.', '$');
+        if (GeneratorUtils.isBlank(packageName)) {
+            return GeneratorUtils.replaceChars(canonicalName, '.', '$');
         } else {
-            String name = StringUtils.substringAfterLast(canonicalName,
+            String name = GeneratorUtils.substringAfterLast(canonicalName,
                     packageName + ".");
             return String.format("%s.%s", packageName,
-                    StringUtils.replaceChars(name, '.', '$'));
+                    GeneratorUtils.replaceChars(name, '.', '$'));
         }
     }
 }

--- a/flow-server/src/main/java/com/vaadin/flow/server/connect/generator/OpenApiSpecGenerator.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/connect/generator/OpenApiSpecGenerator.java
@@ -25,7 +25,6 @@ import io.swagger.v3.core.util.Json;
 import io.swagger.v3.oas.models.OpenAPI;
 import org.apache.commons.configuration2.PropertiesConfiguration;
 import org.apache.commons.io.FileUtils;
-import org.apache.commons.lang3.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
@@ -117,7 +116,7 @@ public class OpenApiSpecGenerator {
             PropertiesConfiguration applicationProperties) {
         String endpoint = applicationProperties.getString(ENDPOINT,
                 DEFAULT_ENDPOINT);
-        String server = StringUtils.removeEnd(
+        String server = GeneratorUtils.removeEnd(
                 applicationProperties.getString(SERVER, DEFAULT_SERVER), "/");
         String serverDescription = applicationProperties
                 .getString(SERVER_DESCRIPTION, DEFAULT_SERVER_DESCRIPTION);

--- a/flow-server/src/main/java/com/vaadin/flow/server/connect/generator/OpenApiSpecGenerator.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/connect/generator/OpenApiSpecGenerator.java
@@ -20,10 +20,10 @@ import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Path;
 import java.util.Collection;
+import java.util.Properties;
 
 import io.swagger.v3.core.util.Json;
 import io.swagger.v3.oas.models.OpenAPI;
-import org.apache.commons.configuration2.PropertiesConfiguration;
 import org.apache.commons.io.FileUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
@@ -59,7 +59,7 @@ public class OpenApiSpecGenerator {
      * @param applicationProperties
      *            the properties with the data required for the generation
      */
-    public OpenApiSpecGenerator(PropertiesConfiguration applicationProperties) {
+    public OpenApiSpecGenerator(Properties applicationProperties) {
         generator = new OpenApiObjectGenerator();
         generator.setOpenApiConfiguration(
                 extractOpenApiConfiguration(applicationProperties));
@@ -113,17 +113,18 @@ public class OpenApiSpecGenerator {
     }
 
     private OpenApiConfiguration extractOpenApiConfiguration(
-            PropertiesConfiguration applicationProperties) {
-        String endpoint = applicationProperties.getString(ENDPOINT,
+            Properties applicationProperties) {
+        String endpoint = (String) applicationProperties.getOrDefault(ENDPOINT,
                 DEFAULT_ENDPOINT);
-        String server = GeneratorUtils.removeEnd(
-                applicationProperties.getString(SERVER, DEFAULT_SERVER), "/");
-        String serverDescription = applicationProperties
-                .getString(SERVER_DESCRIPTION, DEFAULT_SERVER_DESCRIPTION);
-        String applicationTitle = applicationProperties
-                .getString(APPLICATION_TITLE, DEFAULT_APPLICATION_TITLE);
-        String applicationApiVersion = applicationProperties.getString(
-                APPLICATION_API_VERSION, DEFAULT_APPLICATION_API_VERSION);
+        String server = GeneratorUtils.removeEnd((String) applicationProperties
+                .getOrDefault(SERVER, DEFAULT_SERVER), "/");
+        String serverDescription = (String) applicationProperties
+                .getOrDefault(SERVER_DESCRIPTION, DEFAULT_SERVER_DESCRIPTION);
+        String applicationTitle = (String) applicationProperties
+                .getOrDefault(APPLICATION_TITLE, DEFAULT_APPLICATION_TITLE);
+        String applicationApiVersion = (String) applicationProperties
+                .getOrDefault(APPLICATION_API_VERSION,
+                        DEFAULT_APPLICATION_API_VERSION);
         return new OpenApiConfiguration(applicationTitle, applicationApiVersion,
                 server + endpoint, serverDescription);
     }

--- a/flow-server/src/main/java/com/vaadin/flow/server/connect/generator/SchemaResolver.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/connect/generator/SchemaResolver.java
@@ -43,7 +43,6 @@ import io.swagger.v3.oas.models.media.NumberSchema;
 import io.swagger.v3.oas.models.media.ObjectSchema;
 import io.swagger.v3.oas.models.media.Schema;
 import io.swagger.v3.oas.models.media.StringSchema;
-import org.apache.commons.lang3.StringUtils;
 
 class SchemaResolver {
 
@@ -217,8 +216,8 @@ class SchemaResolver {
     }
 
     String getSimpleRef(String ref) {
-        if (StringUtils.contains(ref, SCHEMA_REF_PREFIX)) {
-            return StringUtils.substringAfter(ref, SCHEMA_REF_PREFIX);
+        if (GeneratorUtils.contains(ref, SCHEMA_REF_PREFIX)) {
+            return GeneratorUtils.substringAfter(ref, SCHEMA_REF_PREFIX);
         }
         return ref;
     }

--- a/flow-server/src/main/java/com/vaadin/flow/server/connect/generator/VaadinConnectClientGenerator.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/connect/generator/VaadinConnectClientGenerator.java
@@ -21,9 +21,9 @@ import java.io.IOException;
 import java.io.InputStreamReader;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Path;
+import java.util.Properties;
 import java.util.stream.Collectors;
 
-import org.apache.commons.configuration2.PropertiesConfiguration;
 import org.apache.commons.io.FileUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
@@ -54,8 +54,8 @@ public class VaadinConnectClientGenerator {
      *            the properties with the data required for the generation
      */
     public VaadinConnectClientGenerator(
-            PropertiesConfiguration applicationProperties) {
-        this.serviceEndpoint = applicationProperties.getString(ENDPOINT,
+            Properties applicationProperties) {
+        this.serviceEndpoint = (String)applicationProperties.getOrDefault(ENDPOINT,
                 DEFAULT_ENDPOINT);
     }
 

--- a/flow-server/src/main/java/com/vaadin/flow/server/connect/generator/VaadinConnectTsGenerator.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/connect/generator/VaadinConnectTsGenerator.java
@@ -64,9 +64,6 @@ import io.swagger.v3.parser.core.models.SwaggerParseResult;
 import org.apache.commons.io.FileUtils;
 import org.apache.commons.io.filefilter.AbstractFileFilter;
 import org.apache.commons.io.filefilter.TrueFileFilter;
-import org.apache.commons.lang3.BooleanUtils;
-import org.apache.commons.lang3.ObjectUtils;
-import org.apache.commons.lang3.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
@@ -213,7 +210,7 @@ public class VaadinConnectTsGenerator extends AbstractTypeScriptClientCodegen {
     }
 
     private static String getDefaultClientPath(String path) {
-        path = ObjectUtils.defaultIfNull(path,
+        path = GeneratorUtils.defaultIfNull(path,
                 VaadinConnectClientGenerator.CONNECT_CLIENT_IMPORT_PATH);
         return removeTsExtension(path);
     }
@@ -238,12 +235,13 @@ public class VaadinConnectTsGenerator extends AbstractTypeScriptClientCodegen {
                     .collect(Collectors.toSet());
         } else {
             String error = parseResult == null ? ""
-                    : StringUtils.join(parseResult.getMessages().toArray());
+                    : String.join("", parseResult.getMessages());
             cleanGeneratedFolder(configurator.getOutputDir(),
                     Collections.emptySet());
             throw getUnexpectedOpenAPIException(configurator.getInputSpecURL(),
                     error);
         }
+        
     }
 
     private static void cleanGeneratedFolder(String outputDir,
@@ -409,13 +407,13 @@ public class VaadinConnectTsGenerator extends AbstractTypeScriptClientCodegen {
 
     @Override
     public String toModelFilename(String name) {
-        if (!StringUtils.contains(name, ".")) {
+        if (!GeneratorUtils.contains(name, ".")) {
             return super.toModelFilename(name);
         }
-        String packageName = StringUtils.substringBeforeLast(name, ".");
+        String packageName = GeneratorUtils.substringBeforeLast(name, ".");
         packageName = packageName.replaceAll("\\.", "/");
 
-        String modelName = StringUtils.substringAfterLast(name, ".");
+        String modelName = GeneratorUtils.substringAfterLast(name, ".");
         modelName = super.toModelFilename(modelName);
 
         return packageName + "/" + modelName;
@@ -500,14 +498,14 @@ public class VaadinConnectTsGenerator extends AbstractTypeScriptClientCodegen {
     private String getSimpleNameFromImports(String dataType,
             List<Map<String, String>> imports) {
         for (Map<String, String> anImport : imports) {
-            if (StringUtils.equals(dataType, anImport.get(IMPORT))) {
-                return StringUtils.firstNonBlank(anImport.get("importAs"),
+            if (GeneratorUtils.equals(dataType, anImport.get(IMPORT))) {
+                return GeneratorUtils.firstNonBlank(anImport.get("importAs"),
                         anImport.get("className"));
             }
         }
-        if (StringUtils.contains(dataType, "<")
-                || StringUtils.contains(dataType, "{")
-                || StringUtils.contains(dataType, "|")) {
+        if (GeneratorUtils.contains(dataType, "<")
+                || GeneratorUtils.contains(dataType, "{")
+                || GeneratorUtils.contains(dataType, "|")) {
             return getSimpleNameFromComplexType(dataType, imports);
         }
         return getSimpleNameFromQualifiedName(dataType);
@@ -527,14 +525,14 @@ public class VaadinConnectTsGenerator extends AbstractTypeScriptClientCodegen {
     }
 
     private String getSimpleNameFromQualifiedName(String qualifiedName) {
-        if (StringUtils.contains(qualifiedName, ".")) {
-            return StringUtils.substringAfterLast(qualifiedName, ".");
+        if (GeneratorUtils.contains(qualifiedName, ".")) {
+            return GeneratorUtils.substringAfterLast(qualifiedName, ".");
         }
         return qualifiedName;
     }
 
     private String convertQualifiedNameToModelPath(String qualifiedName) {
-        return "./" + StringUtils.replaceChars(qualifiedName, '.', '/');
+        return "./" + GeneratorUtils.replaceChars(qualifiedName, '.', '/');
     }
 
     private void validateOperationTags(String path, String httpMethod,
@@ -586,7 +584,7 @@ public class VaadinConnectTsGenerator extends AbstractTypeScriptClientCodegen {
 
     @Override
     protected void addImport(CodegenModel m, String type) {
-        if (!StringUtils.equals(m.getName(), type)) {
+        if (!GeneratorUtils.equals(m.getName(), type)) {
             super.addImport(m, type);
         }
     }
@@ -639,7 +637,7 @@ public class VaadinConnectTsGenerator extends AbstractTypeScriptClientCodegen {
         String modelFilePath = convertQualifiedNameToModelPath(
                 qualifiedNameForRelative);
         // Remove the class name, only consider the parent folder
-        modelFilePath = StringUtils.substringBeforeLast(modelFilePath, "/");
+        modelFilePath = GeneratorUtils.substringBeforeLast(modelFilePath, "/");
         adjustImportInformation(imports, modelFilePath);
     }
 
@@ -657,7 +655,7 @@ public class VaadinConnectTsGenerator extends AbstractTypeScriptClientCodegen {
         Set<String> usedNames = new HashSet<>();
         // Make sure the import list are always in the same orders in when
         // generating different times.
-        imports.sort((o1, o2) -> StringUtils.compare((String) o1.get(IMPORT),
+        imports.sort((o1, o2) -> GeneratorUtils.compare((String) o1.get(IMPORT),
                 (String) o2.get(IMPORT)));
         for (Map<String, Object> anImport : imports) {
             String importQualifiedName = (String) anImport.get(IMPORT);
@@ -686,18 +684,19 @@ public class VaadinConnectTsGenerator extends AbstractTypeScriptClientCodegen {
     private String relativePathToNodeImport(String relativePath) {
         // on Windows Node imports should still use Unix-style path separator
         relativePath = relativePath.replace("\\", "/");
-        return StringUtils.prependIfMissing(relativePath, "./", ".", "/");
+        // prepend with `./` if the string does not start with `./`, `.` or `/`
+        return relativePath.replaceFirst("^(?!(\\./|\\.|/))", "./");
     }
 
     private String getUniqueNameFromQualifiedName(Set<String> usedNames,
             String qualifiedName) {
-        String[] packageSegments = StringUtils.split(qualifiedName, '.');
+        String[] packageSegments = qualifiedName == null ? null : qualifiedName.split("\\.");
         StringBuilder classNameBuilder = new StringBuilder();
         String newClassName = "";
         if (packageSegments != null && packageSegments.length > 1) {
             for (int i = packageSegments.length - 1; i >= 0; i--) {
                 classNameBuilder.insert(0,
-                        StringUtils.capitalize(packageSegments[i]));
+                        GeneratorUtils.capitalize(packageSegments[i]));
                 newClassName = classNameBuilder.toString();
                 if (!usedNames.contains(newClassName)) {
                     return newClassName;
@@ -716,7 +715,7 @@ public class VaadinConnectTsGenerator extends AbstractTypeScriptClientCodegen {
 
     private void setShouldShowTsDoc(List<CodegenOperation> operations) {
         for (CodegenOperation coop : operations) {
-            boolean hasDescription = StringUtils.isNotBlank(coop.getNotes());
+            boolean hasDescription = GeneratorUtils.isNotBlank(coop.getNotes());
             boolean hasParameter = hasParameterDescription(coop);
             boolean hasResponseDescription = hasResponseDescription(coop);
             if (hasDescription || hasParameter || hasResponseDescription) {
@@ -728,7 +727,7 @@ public class VaadinConnectTsGenerator extends AbstractTypeScriptClientCodegen {
 
     private boolean hasResponseDescription(CodegenOperation coop) {
         for (CodegenResponse response : coop.getResponses()) {
-            if (StringUtils.isNotBlank(response.getMessage())) {
+            if (GeneratorUtils.isNotBlank(response.getMessage())) {
                 return true;
             }
         }
@@ -741,7 +740,7 @@ public class VaadinConnectTsGenerator extends AbstractTypeScriptClientCodegen {
                     .getVendorExtensions()
                     .get(EXTENSION_VAADIN_CONNECT_PARAMETERS);
             if (parametersList != null && parametersList.stream()
-                    .anyMatch(parameterInformation -> StringUtils.isNotBlank(
+                    .anyMatch(parameterInformation -> GeneratorUtils.isNotBlank(
                             parameterInformation.getDescription()))) {
                 return true;
             }
@@ -781,7 +780,7 @@ public class VaadinConnectTsGenerator extends AbstractTypeScriptClientCodegen {
 
     private Set<String> collectImportsFromSchema(Schema schema) {
         Set<String> imports = new HashSet<>();
-        if (StringUtils.isNotBlank(schema.get$ref())) {
+        if (GeneratorUtils.isNotBlank(schema.get$ref())) {
             imports.add(getSimpleRef(schema.get$ref()));
         }
         if (schema instanceof ArraySchema) {
@@ -815,7 +814,7 @@ public class VaadinConnectTsGenerator extends AbstractTypeScriptClientCodegen {
             name = isReservedWord(name) ? escapeReservedWord(name) : name;
             String type = getTypeDeclaration(entry.getValue());
             String description = entry.getValue().getDescription();
-            if (StringUtils.isBlank(description)) {
+            if (GeneratorUtils.isBlank(description)) {
                 description = getDescriptionFromParameterExtension(name,
                         requestSchema);
             }
@@ -840,7 +839,7 @@ public class VaadinConnectTsGenerator extends AbstractTypeScriptClientCodegen {
     @Override
     public String getTypeDeclaration(Schema schema) {
         String optionalSuffix = "";
-        if (BooleanUtils.isTrue(schema.getNullable())) {
+        if (GeneratorUtils.isTrue(schema.getNullable())) {
             optionalSuffix = OPTIONAL_SUFFIX;
         }
         if (schema instanceof ArraySchema) {
@@ -848,7 +847,7 @@ public class VaadinConnectTsGenerator extends AbstractTypeScriptClientCodegen {
             Schema inner = arraySchema.getItems();
             return String.format("Array<%s>%s", this.getTypeDeclaration(inner),
                     optionalSuffix);
-        } else if (StringUtils.isNotBlank(schema.get$ref())) {
+        } else if (GeneratorUtils.isNotBlank(schema.get$ref())) {
             return getSimpleRef(schema.get$ref()) + optionalSuffix;
         } else if (schema.getAdditionalProperties() != null) {
             Schema inner = (Schema) schema.getAdditionalProperties();

--- a/flow-server/src/main/java/com/vaadin/flow/server/frontend/AbstractTaskConnectGenerator.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/frontend/AbstractTaskConnectGenerator.java
@@ -20,9 +20,8 @@ import java.io.File;
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
+import java.util.Properties;
 
-import org.apache.commons.configuration2.PropertiesConfiguration;
-import org.apache.commons.configuration2.ex.ConfigurationException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
@@ -38,15 +37,14 @@ abstract class AbstractTaskConnectGenerator implements FallibleCommand {
         this.applicationProperties = applicationProperties;
     }
 
-    protected PropertiesConfiguration readApplicationProperties() {
-        PropertiesConfiguration config = new PropertiesConfiguration();
+    protected Properties readApplicationProperties() {
+        Properties config = new Properties();
 
         if (applicationProperties != null && applicationProperties.exists()) {
-            try {
-                BufferedReader bufferedReader = Files.newBufferedReader(
-                        applicationProperties.toPath(), StandardCharsets.UTF_8);
-                config.read(bufferedReader);
-            } catch (IOException | ConfigurationException e) {
+            try (BufferedReader bufferedReader = Files.newBufferedReader(
+                    applicationProperties.toPath(), StandardCharsets.UTF_8)) {
+                config.load(bufferedReader);
+            } catch (IOException e) {
                 log().info(String.format(
                         "Can't read the application"
                                 + ".properties file from %s",

--- a/flow-server/src/test/java/com/vaadin/flow/server/connect/generator/TestUtils.java
+++ b/flow-server/src/test/java/com/vaadin/flow/server/connect/generator/TestUtils.java
@@ -21,18 +21,17 @@ import java.io.FileReader;
 import java.io.IOException;
 import java.io.InputStreamReader;
 import java.net.URL;
+import java.util.Properties;
 import java.util.stream.Collectors;
-
-import org.apache.commons.configuration2.PropertiesConfiguration;
 
 public final class TestUtils {
     private TestUtils() {
     }
 
-    public static PropertiesConfiguration readProperties(String filePath) {
-        PropertiesConfiguration properties = new PropertiesConfiguration();
+    public static Properties readProperties(String filePath) {
+        Properties properties = new Properties();
         try {
-            properties.read(new FileReader(filePath));
+            properties.load(new FileReader(filePath));
         } catch (Exception e) {
             throw new AssertionError(String.format(
                     "Failed to read the properties file '%s", filePath));

--- a/flow-server/src/test/java/com/vaadin/flow/server/connect/generator/VaadinConnectClientGeneratorTest.java
+++ b/flow-server/src/test/java/com/vaadin/flow/server/connect/generator/VaadinConnectClientGeneratorTest.java
@@ -19,8 +19,8 @@ import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.Properties;
 
-import org.apache.commons.configuration2.PropertiesConfiguration;
 import org.apache.commons.lang3.StringUtils;
 import org.junit.Assert;
 import org.junit.Before;
@@ -44,7 +44,7 @@ public class VaadinConnectClientGeneratorTest {
     public void should_GenerateConnectClientDefault_When_NoApplicationPropertiesInput()
             throws Exception {
         VaadinConnectClientGenerator generator = new VaadinConnectClientGenerator(
-                new PropertiesConfiguration());
+                new Properties());
 
         generator.generateVaadinConnectClientFile(outputPath);
 

--- a/flow-server/src/test/java/com/vaadin/flow/server/connect/generator/openapi/OpenApiSpecBasedTests.java
+++ b/flow-server/src/test/java/com/vaadin/flow/server/connect/generator/openapi/OpenApiSpecBasedTests.java
@@ -23,8 +23,8 @@ import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.Properties;
 
-import org.apache.commons.configuration2.PropertiesConfiguration;
 import org.apache.commons.lang3.StringUtils;
 import org.junit.Assert;
 import org.junit.BeforeClass;
@@ -114,7 +114,7 @@ public class OpenApiSpecBasedTests {
                 outputDirectory.getRoot().getAbsolutePath(),
                 VaadinConnectClientGenerator.CONNECT_CLIENT_NAME);
         VaadinConnectClientGenerator vaadinConnectClientGenerator = new VaadinConnectClientGenerator(
-                new PropertiesConfiguration());
+                new Properties());
         // First generating round
         vaadinConnectClientGenerator
                 .generateVaadinConnectClientFile(defaultConnectClient);

--- a/flow-server/src/test/java/com/vaadin/flow/server/connect/generator/services/AbstractServiceGenerationTest.java
+++ b/flow-server/src/test/java/com/vaadin/flow/server/connect/generator/services/AbstractServiceGenerationTest.java
@@ -18,6 +18,7 @@ package com.vaadin.flow.server.connect.generator.services;
 
 import javax.annotation.Nullable;
 import javax.annotation.security.DenyAll;
+
 import java.io.File;
 import java.io.IOException;
 import java.lang.reflect.Field;
@@ -42,6 +43,7 @@ import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 import java.util.Optional;
+import java.util.Properties;
 import java.util.Set;
 
 import com.fasterxml.jackson.annotation.JsonIgnore;
@@ -64,13 +66,8 @@ import io.swagger.v3.oas.models.media.StringSchema;
 import io.swagger.v3.oas.models.responses.ApiResponse;
 import io.swagger.v3.oas.models.responses.ApiResponses;
 import io.swagger.v3.parser.OpenAPIV3Parser;
-import org.apache.commons.configuration2.PropertiesConfiguration;
 import org.apache.commons.lang3.StringUtils;
 import org.junit.Assert;
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertNull;
-import static org.junit.Assert.assertTrue;
 import org.junit.Before;
 import org.junit.Rule;
 import org.junit.rules.TemporaryFolder;
@@ -83,6 +80,11 @@ import com.vaadin.flow.server.connect.generator.OpenApiObjectGenerator;
 import com.vaadin.flow.server.connect.generator.OpenApiSpecGenerator;
 import com.vaadin.flow.server.connect.generator.TestUtils;
 import com.vaadin.flow.server.connect.generator.VaadinConnectTsGenerator;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
 
 public abstract class AbstractServiceGenerationTest {
     private static final List<Class<?>> JSON_NUMBER_CLASSES = Arrays.asList(
@@ -153,8 +155,8 @@ public abstract class AbstractServiceGenerationTest {
 
     private void generateAndVerify(URL customApplicationProperties,
             URL expectedOpenApiJsonResourceUrl) {
-        PropertiesConfiguration applicationProperties = customApplicationProperties == null
-                ? new PropertiesConfiguration()
+        Properties applicationProperties = customApplicationProperties == null
+                ? new Properties()
                 : TestUtils
                         .readProperties(customApplicationProperties.getPath());
         new OpenApiSpecGenerator(applicationProperties).generateOpenApiSpec(
@@ -565,7 +567,7 @@ public abstract class AbstractServiceGenerationTest {
      * .ts</code>. For example: The generated model of
      * <code>com.vaadin.flow.server.connect.SomeModel</code> will be expected to
      * be the same as file <code>expected-model-connect.SomeModel.ts</code>
-     * 
+     *
      * @param expectedClass
      * @return
      */

--- a/flow-server/src/test/java/com/vaadin/flow/server/connect/generator/services/json/JsonTestServiceGeneratedTest.java
+++ b/flow-server/src/test/java/com/vaadin/flow/server/connect/generator/services/json/JsonTestServiceGeneratedTest.java
@@ -19,18 +19,18 @@ package com.vaadin.flow.server.connect.generator.services.json;
 import java.io.File;
 import java.util.Arrays;
 import java.util.Collections;
+import java.util.Properties;
 
 import com.fasterxml.jackson.core.Version;
-import org.apache.commons.configuration2.PropertiesConfiguration;
 import org.junit.Test;
-
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
 
 import com.vaadin.flow.server.connect.generator.OpenApiSpecGenerator;
 import com.vaadin.flow.server.connect.generator.VaadinConnectClientGenerator;
 import com.vaadin.flow.server.connect.generator.VaadinConnectTsGenerator;
 import com.vaadin.flow.server.connect.generator.services.AbstractServiceGenerationTest;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
 
 public class JsonTestServiceGeneratedTest
     extends AbstractServiceGenerationTest {
@@ -66,7 +66,7 @@ public class JsonTestServiceGeneratedTest
     String expectedImport = String.format("import client from '%s';",
         customConnectClientPath);
 
-    new OpenApiSpecGenerator(new PropertiesConfiguration()).generateOpenApiSpec(
+    new OpenApiSpecGenerator(new Properties()).generateOpenApiSpec(
         Collections
             .singletonList(java.nio.file.Paths.get("src/test/java", getClass()
                 .getPackage().getName().replace('.', File.separatorChar))),

